### PR TITLE
Chef Sample App - yaml configuration file

### PR DESCRIPTION
**Pull request by rochaferraz**
* Changed chef sample app configuration file to yaml. It picks up a defaults from environment variables first time the script is run
* Builds on current tot for esp32, nrfconnect and linux targets
* nrf52840dk_nrf52840 renamed to nrfconnect

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: .github/.wordlist.txt [hunk 1]
```diff
@@ -1327,6 +1327,7 @@ ureverse
 URI
 urlrequest
 usbmodem
+usbmodemXXXXXXX
```
Comment (bzbarsky-apple) [inline .github/.wordlist.txt:1330]: You should be able to avoid these things now if the md clearly marks these as code blocks using backticks.

**Review by bzbarsky-apple — APPROVED** [review]

File: .github/.wordlist.txt (hunk 1 above)
Comment (rochaferraz) [reply .github/.wordlist.txt:1330]: ok, thanks!

Comment (andy31415) [reply .github/.wordlist.txt:1330]: Please remove the non-words from wordlist and use only backticks for code (and that should make spellcheck happy)

File: .github/.wordlist.txt [hunk 2]
```diff
@@ -1418,6 +1418,9 @@ xFFFF
 xfffff
 xtensa
 xwayland
+xxxxx
+XXXX
+XXXXXXXX
```
Comment (andy31415) [inline .github/.wordlist.txt:1423]: `/XX+/` are not words and should be not in wordlist. Use backticks in md to make spellcheck happy (and get better format in general)

**Pull request merged**